### PR TITLE
add sepp test and metrics for edge cache performance

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* Adds a workload `EdgeCache` to sepp.
  The test can be run as follows:
```
for l in 0.04; do
  for u in 0.25 0.33 0.4 0.5 0.66; do
    for m in 100 1048576000; do
      for r in 2 3 5; do
        rm -rf temp
        echo "lower: $l, upper: $u, min: $m, reads: $r"
        build/bin/sepp sepp.json -- databaseDirectory=\"temp\" cache.idealLowerFillRatio=$l cache.idealUpperFillRatio=$u cache.minValueSizeForEdgeCompression=$m workload.edgeCache.default.readsPerEdge=$r
      done
    done
  done
done
```
Sample config file:
```
{
  "cache": {
    "idealLowerFillRatio": 0.04,
    "idealUpperFillRatio": 0.25,
    "minValueSizeForEdgeCompression": 1048576000,
    "accelerationFactorForEdgeCompression": 1,
    "maxSpareAllocation": 0
  },
  "setup": {
    "collections": [
      {
        "name": "testcol",
        "type": "edge"
      }
    ]
  },
  "workload": {
    "edgeCache": {
      "default": {
        "collection": "testcol",
        "documentsPerTrx": 100,
        "edgesPerVertex": 10,
        "readsPerEdge": 3
      },
      "stopAfter": {
        "runtime": 300000
      },
      "threads": 1
    }
  }
}
```

* Added the following metrics for the in-memory edge cache:

  - `rocksdb_cache_edge_inserts_total`: total number of insertions into the
    in-memory edge cache.
  - `rocksdb_cache_edge_compressed_inserts_total`: total number of insertions
    into the in-memory edge cache that used compression.

* Added the startup option `--cache.max-spare-memory-usage` to control memory
  usage for spare, unused hash tables in the in-memory caching subsystem. This
  option can be used to cap the memory used by spare tables. It can be set to
  a value of 0 to not use any memory except for active hash tables. 

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: -
  - [ ] Backport for 3.10: -
  - [ ] Backport for 3.9: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by mchacki — APPROVED** [review]
LGTM.
I like this new test.

I only have minor (almost nitpicking) comments.
Take or drop as you prefer.

I am just wondering:
This test only goes outbound, but not inbound. (which is symmetric implementation)
Shall we add a future test for inbound? Or more important an "any" case to see if/how the two caches in use interfere?

File: arangod/Cache/Manager.h [hunk 1]
```diff
@@ -276,7 +273,8 @@ class Manager {
   template<typename Hasher>
   friend class TransactionalCache;
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply arangod/Cache/Manager.h:274]: No, I did not remove all private members of the `Manager` class. It still has several.
I only combined a few existing private members from individual members into a (private) CacheOptions member.
So friends will be friends.

File: arangod/Cache/CacheOptionsFeature.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+  options
+      ->addOption("--cache.size",
+                  "The global size limit for all caches (in bytes).",
+                  new UInt64Parameter(&_options.cacheSize),
+                  arangodb::options::makeDefaultFlags(
+                      arangodb::options::Flags::Dynamic))
+      .setLongDescription(R"(The global caching system, all caches, and all the
+data contained therein are constrained to this limit.
+
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply arangod/Cache/CacheOptionsFeature.cpp:67]: Yes, that blank line is probably intentional. The code here existed before in `CacheManagerFeature.cpp`, and I only moved it to a new location (`CacheOptionsFeature.cpp`), but did not modify it.

File: arangod/Cache/Manager.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (mchacki) [reply arangod/Cache/Manager.h:274]: Agreed there is one I missed
